sources/VT100Screen.m

```diff
@@ -4277,6 +4288,22 @@ - (void)dumpScreen
     NSLog(@"%@", [self debugString]);
 }
 
+- (void)linkRun:(VT100GridRun)run
+       withCode:(unsigned short)code {
+    screen_char_t fg = { 0 };
+    screen_char_t bg = { 0 };
+    
+    fg.urlCode = code;
+    
+    for (NSValue *value in [currentGrid_ rectsForRun:run]) {
+        VT100GridRect rect = [value gridRectValue];
+        [currentGrid_ setBackgroundColor:bg
```

Comment:
This will have the undesirable side-effect of changing the background and foreground colors when all you want is to set the URL code. It would make sense to add a new method to `VT100Grid` like `setURLCode:inRectFrom:to:`.

Reply:
@gnachman you're absolutely right… I did not thought about this. I've followed your advice with 7279776.